### ISSUE TITLE
Schema-only validations

### ISSUE THREAD

**Issue by benasher44** [sequelize/sequelize#15919]
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

### Describe the feature you'd like to see implemented

Allow flagging that the validation should only be checked by the database (schema), for those validations that are backed by schema constraints.

### Describe why you would like this feature to be added to Sequelize

We want to be able to say that a column does not allow null, but the column is filled out by a trigger in the database. At present, there's no way to communicate this to sequelize. If you create a sequelize model and attempt to save it, such columns will fail validations. If we could tell sequelize to skip the code-level validation, we can communicate that we're going to lean on the database schema-level validation to cover it.

### Is this feature dialect-specific?

- [x] No. This feature is relevant to Sequelize as a whole.
- [ ] Yes. This feature only applies to the following dialect(s): 

### Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in implementing my feature.

---

*Indicate your interest in the addition of this feature by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by ephys** [comment]
You can disable js-side validation for a given DB call by setting the `validate` option to false

This is poorly typed and documented but you can also use the `skip` option to skip validation for a specific attribute:

```ts
User.create({
  // data
}, {
  validate: false,
  // or
  skip: ['name-of-skipped-attribute'],
});
```

**Comment by benasher44** [comment]
Ah yep sorry I should have called out that I found this one. We have various save/create calls everywhere, and remembering to do this for a given model is painful. We're using sequelize-typescript, so ideally we could disable it generally

**Comment by benasher44** [comment]
I came up with a different solution here. What I did is set a flag that enables the nonnull constraint during database migrations, so that the nonnull constraint schema migration is created. That flag is not set at runtime, which disables sequelize's nonnull checks. This solution works for me and gets the intended behavior we're looking for. I'm happy to close this, and it could be reopened if others decide this is useful enough to benefit more than just me and my team.
